source/extensions/filters/network/thrift_proxy/decoder_events.h

```diff
@@ -17,6 +17,32 @@ enum class FilterStatus {
   StopIteration
 };
 
+enum class DecoderEvent {
+  TransportBegin,
+  TransportEnd,
+  PassthroughData,
+  MessageBegin,
+  MessageEnd,
+  StructBegin,
+  StructEnd,
+  FieldBegin,
+  FieldEnd,
+  BoolValue,
+  ByteValue,
+  DoubleValue,
+  Int16Value,
+  Int32Value,
+  Int64Value,
+  StringValue,
+  ListBegin,
+  ListEnd,
+  SetBegin,
+  SetEnd,
+  MapBegin,
+  MapEnd,
+  Default
```

Comment:
Hmm `Default` doesn't say much -- can we come up with a better name? Or at least at a comment as to how this state is meant to be used?

Reply:
Great catch! Renaming makes sense. Rename to ContinueDecode and add some comments as well